server/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -613,49 +611,25 @@ public XContentBuilder value(byte[] value, int offset, int length) throws IOExce
     }
 
     /**
-     * Writes the binary content of the given {@link BytesRef}.
-     *
-     * Use {@link org.elasticsearch.common.xcontent.XContentParser#binaryValue()} to read the value back
-     */
-    public XContentBuilder field(String name, BytesRef value) throws IOException {
-        return field(name).binaryValue(value);
-    }
-
-    /**
-     * Writes the binary content of the given {@link BytesRef} as UTF-8 bytes.
+     * Writes the binary content of the given byte array as UTF-8 bytes.
      *
      * Use {@link XContentParser#charBuffer()} to read the value back
      */
-    public XContentBuilder utf8Field(String name, BytesRef value) throws IOException {
-        return field(name).utf8Value(value);
-    }
-
-    /**
-     * Writes the binary content of the given {@link BytesRef}.
-     *
-     * Use {@link org.elasticsearch.common.xcontent.XContentParser#binaryValue()} to read the value back
-     */
-    public XContentBuilder binaryValue(BytesRef value) throws IOException {
-        if (value == null) {
-            return nullValue();
-        }
-        value(value.bytes, value.offset, value.length);
-        return this;
+    public XContentBuilder utf8Field(String name, byte[] bytes, int offset, int length) throws IOException {
```

Comment:
Maybe as a followup (at sometime, not that important) the callers of this could be made to use the one line impl here. Just seems like a silly method to maintain.

Reply:
Yep, in fact, looking at the callers, this is only ever used in... 1 test, and 1 javadoc, so I'll remove it in the next sweep of things in XContentBuilder